examples/esp32-with-cred/esp32-server-with-cred.rs

```diff
@@ -0,0 +1,246 @@
+    robot_srv_der_key: Vec<u8>,
+}
+
+impl NvsStaticVars {
+    fn new() -> Result<NvsStaticVars, ServerError> {
+        let nvs = EspDefaultNvsPartition::take()?;
+        info!("get namespace...");
+        let viam_nvs = EspNvs::new(nvs.clone(), VIAM_NVS_NAMESPACE, true)?;
+        info!("loading creds...");
+        let res = Ok(NvsStaticVars {
+            wifi_ssid: get_str_from_nvs(&viam_nvs, "WIFI_SSID")?,
+            wifi_pwd: get_str_from_nvs(&viam_nvs, "WIFI_PASSWORD")?,
+            robot_secret: get_str_from_nvs(&viam_nvs, "ROBOT_SECRET")?,
+            robot_id: get_str_from_nvs(&viam_nvs, "ROBOT_ID")?,
+            robot_dtls_cert: get_blob_from_nvs(&viam_nvs, "ROBOT_DTLS_CERT")?,
+            robot_dtls_key_pair: get_blob_from_nvs(&viam_nvs, "DTLS_KEY_PAIR")?,
+            robot_dtls_cert_fp: get_str_from_nvs(&viam_nvs, "DTLS_CERT_FP")?,
+            robot_srv_pem_chain: get_blob_from_nvs(&viam_nvs, "SRV_PEM_CHAIN")?,
+            robot_srv_pem_ca: get_blob_from_nvs(&viam_nvs, "CA_CRT")?,
+            robot_srv_der_key: get_blob_from_nvs(&viam_nvs, "SRV_DER_KEY")?,
+        });
+        res
+    }
+}
+
+fn main() -> Result<(), ServerError> {
+    std::env::set_var("RUST_BACKTRACE", "1");
+    esp_idf_sys::link_patches();
+
+    esp_idf_svc::log::EspLogger::initialize_default();
+    let sys_loop_stack = EspSystemEventLoop::take()?;
+
+    #[cfg(not(feature = "qemu"))]
+    let periph = Peripherals::take().ok_or(ServerError::PeripheralsError)?;
+
+    #[cfg(feature = "qemu")]
+    let repr = {
+        let board = Arc::new(Mutex::new(FakeBoard::new(vec![])));
+        let mut res: ResourceMap = HashMap::with_capacity(1);
+        res.insert(
+            ResourceName {
+                namespace: "rdk".to_string(),
+                r#type: "component".to_string(),
+                subtype: "board".to_string(),
+                name: "b".to_string(),
+            },
+            ResourceType::Board(board),
+        );
+        RobotRepresentation::WithRobot(LocalRobot::new(res))
+    };
+    #[cfg(not(feature = "qemu"))]
+    let repr = RobotRepresentation::WithRegistry(ComponentRegistry::default());
+
+    {
+        esp_idf_sys::esp!(unsafe {
+            esp_idf_sys::esp_vfs_eventfd_register(&esp_idf_sys::esp_vfs_eventfd_config_t {
+                max_fds: 5,
+            })
+        })?;
+    }
+
+    #[cfg(feature = "qemu")]
+    let (ip, _block_eth) = {
+        use esp_idf_hal::prelude::Peripherals;
+        info!("creating eth object");
+        let mut eth = Box::new(esp_idf_svc::eth::EspEth::wrap(
+            esp_idf_svc::eth::EthDriver::new_openeth(
+                Peripherals::take().ok_or(ServerError::PeripheralsError).mac,
+                sys_loop_stack.clone(),
+            )?,
+        )?);
+        let _ = eth_configure(&sys_loop_stack, &mut eth)?;
+        let ip = Ipv4Addr::new(10, 1, 12, 187);
+        (ip, eth)
+    };
+
+    #[cfg(not(feature = "qemu"))]
+    info!("load vars...");
```

Comment:
...if I'm reading this correctly, we only log this line if we're not using qemu. That feels like an accident; were we supposed to log this line no matter what and do something else only when not qemu? or maybe I read it wrong?

Reply:
the flag shouldn't be there at all actually, I'll remove